feat: Adds async reactor documentation (#76)
Basis-Theory/developers.basistheory.com: docs/concepts/what-are-reactors.mdx

  }'
```

> ## Asynchronous Reactors

Author: ```suggestion
### Asynchronous Reactors
```

> Reactors can be invoked asynchronously by providing a `callback_url` parameter within the [request](/docs/api/reactors#invoke-a-reactor).

Reviewer: ```suggestion
Reactors are invoked asynchronously by providing a `callback_url` parameter within the [request](/docs/api/reactors#invoke-a-reactor).
```

The reactor will perform some synchronous validation to ensure the request is valid,
then immediately respond with an empty HTTP response with `202 Accepted` status code.

Your reactor code will then be asynchronously executed in our serverless compute environment for up to the configured timeout specified in the `timeout_ms`
parameter. If not specified, this value defaults to `10000` (i.e. 10 seconds). If the timeout period is exceeded, the reactor will be terminated and you will receive
a callback containing a [Reactor Runtime Error](/docs/api/reactors/reactor-errors#error-types) indicating that a timeout occurred.

Once the reactor is completed, the response from the reactor code
will be delivered in the body of an HTTP `POST` request to your `callback_url`.

Your callback endpoint must be hosted using HTTPS, and should respond with a `2xx` status code after receiving the message. If your servers fail to

> respond with a `2xx` status code, delivery will retried up to 10 times with exponential backoff.

Reviewer 2: should we specify what this exponential backoff is?

Reviewer 2: or maybe just provide the range it could be within

Author: > delivery will retried up to 10 times with exponential backoff over the next ~2.5 hours
